src/solver/optimisation/constraints/PowerOutputVariation.cpp

```diff
@@ -0,0 +1,47 @@
+#include "PowerOutputVariation.h"
+
+void PowerOutputVariation::add(int pays, int cluster, int clusterIndex, int pdt, bool Simulation)
+{
+    if (!Simulation)
+    {
+        const PALIERS_THERMIQUES& PaliersThermiquesDuPays = problemeHebdo->PaliersThermiquesDuPays[pays];
+        double maxUpwardPowerRampingRate = PaliersThermiquesDuPays.maxUpwardPowerRampingRate[clusterIndex];
+        double pminDUnGroupeDuPalierThermique = PaliersThermiquesDuPays.pminDUnGroupeDuPalierThermique[clusterIndex];
+        // constraint : P(t) - P(t-1) - l * M^+(t) - P^+ + P^- = 0
+        if (pdt > 0)
+        {
+            builder.updateHourWithinWeek(pdt)
+              .DispatchableProduction(cluster, 1.0)
+              .DispatchableProduction(cluster, -1.0, -1, problemeHebdo->NombreDePasDeTempsPourUneOptimisation)
+              .NumberStartingDispatchableUnits(cluster, -pminDUnGroupeDuPalierThermique)
+              .ProductionIncreaseAboveMin(cluster, -1.0)
+              .ProductionDecreaseAboveMin(cluster, 1.0)
+              .equalTo();
+        }
+        else
+        {
+            builder.updateHourWithinWeek(pdt)
+              .DispatchableProduction(cluster, 1.0)
+              .NumberStartingDispatchableUnits(cluster, -pminDUnGroupeDuPalierThermique)
+              .ProductionIncreaseAboveMin(cluster, -1.0)
+              .ProductionDecreaseAboveMin(cluster, 1.0)
+              .equalTo();
+        }
+        if (builder.NumberOfVariables() > 0)
+        {
+            ConstraintNamer namer(problemeHebdo->ProblemeAResoudre->NomDesContraintes);
+
+            namer.UpdateTimeStep(problemeHebdo->weekInTheYear * 168 + pdt);
+            namer.UpdateArea(problemeHebdo->NomsDesPays[pays]);
+
+            namer.ProductionOutputVariation(problemeHebdo->ProblemeAResoudre->NombreDeContraintes, PaliersThermiquesDuPays.NomsDesPaliersThermiques[clusterIndex]);
+        }
+        builder.build();
+    }
+    else
+    {
+        int add = (pdt == 0) ? 4 : 5;
+        problemeHebdo->NbTermesContraintesPourLesRampes += add;
+        problemeHebdo->ProblemeAResoudre->NombreDeContraintes++;
+    }
```

Comment:
To be checked carefully

Reply:
with commit [624f2ce](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1817/commits/624f2ce5436532ef019be957135c5cbb154c5a60), the ramping constraints become cyclical (the evolution of the production for the first hour is constrained the production of the last hour). Then, we don't need to use a constraint for the first time step that is different from the other time steps anymore. This code is now removed.